From da6f7a99e25c9fa0fca75576eaf29f0e8fe67441 Mon Sep 17 00:00:00 2001
From: Brent Cook <busterb@gmail.com>
Date: Tue, 30 Dec 2014 08:59:19 -0600
Subject: [PATCH 02/12] removed unused libevent header and structures

---
 src/usr.sbin/ntpd/ntpd.h | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/src/usr.sbin/ntpd/ntpd.h b/src/usr.sbin/ntpd/ntpd.h
index 8095be3..aa9858b 100644
--- a/src/usr.sbin/ntpd/ntpd.h
+++ b/src/usr.sbin/ntpd/ntpd.h
@@ -28,7 +28,6 @@
 #include <netdb.h>
 #include <pwd.h>
 #include <stdarg.h>
-#include <event.h>
 #include <poll.h>
 
 #include "ntp.h"
@@ -190,14 +189,6 @@ struct ntpd_conf {
 	u_int8_t					filters;
 };
 
-struct imsgev {
-	struct imsgbuf		 ibuf;
-	void			(*handler)(int, short, void *);
-	struct event		 ev;
-	void			*data;
-	short			 events;
-};
-
 struct ctl_show_status {
 	u_int		 peercnt;
 	u_int		 sensorcnt;
@@ -240,7 +231,6 @@ enum blockmodes {
 
 struct ctl_conn {
 	TAILQ_ENTRY(ctl_conn)	entry;
-	struct imsgev		iev;
 	struct imsgbuf		ibuf;
 };
 
-- 
1.9.1

