From 6c6ea48c43c8438d0fadff1d994977f270f6236a Mon Sep 17 00:00:00 2001
From: Brent Cook <busterb@gmail.com>
Date: Mon, 4 May 2015 04:27:29 -0500
Subject: [PATCH 10/13] add a method for updating the realtime clock on sync

from Christian Weisgerber
---
 src/usr.sbin/ntpd/ntpd.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/usr.sbin/ntpd/ntpd.c b/src/usr.sbin/ntpd/ntpd.c
index 57f1f4ff3..0eb7f573c 100644
--- a/src/usr.sbin/ntpd/ntpd.c
+++ b/src/usr.sbin/ntpd/ntpd.c
@@ -55,6 +55,7 @@ const char     *ctl_lookup_option(char *, const char **);
 void		show_status_msg(struct imsg *);
 void		show_peer_msg(struct imsg *, int);
 void		show_sensor_msg(struct imsg *, int);
+void		update_time_sync_status(int);
 
 volatile sig_atomic_t	 quit = 0;
 volatile sig_atomic_t	 reconfig = 0;
@@ -482,6 +483,7 @@ ntpd_adjtime(double d)
 	else if (!firstadj && olddelta.tv_sec == 0 && olddelta.tv_usec == 0)
 		synced = 1;
 	firstadj = 0;
+	update_time_sync_status(synced);
 	return (synced);
 }
 
-- 
2.13.0

