From 0ae8358adca5935a76b908938104ef9cb1ee60c1 Mon Sep 17 00:00:00 2001
From: Brent Cook <busterb@gmail.com>
Date: Mon, 4 May 2015 04:27:29 -0500
Subject: [PATCH 12/12] add a method for updating the realtime clock on sync

from Christian Weisgerber
---
 src/usr.sbin/ntpd/ntpd.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/usr.sbin/ntpd/ntpd.c b/src/usr.sbin/ntpd/ntpd.c
index 310e808..2f89aed 100644
--- a/src/usr.sbin/ntpd/ntpd.c
+++ b/src/usr.sbin/ntpd/ntpd.c
@@ -53,6 +53,7 @@ const char     *ctl_lookup_option(char *, const char **);
 void		show_status_msg(struct imsg *);
 void		show_peer_msg(struct imsg *, int);
 void		show_sensor_msg(struct imsg *, int);
+void		update_time_sync_status(int);
 
 volatile sig_atomic_t	 quit = 0;
 volatile sig_atomic_t	 reconfig = 0;
@@ -423,6 +424,7 @@ ntpd_adjtime(double d)
 	else if (!firstadj && olddelta.tv_sec == 0 && olddelta.tv_usec == 0)
 		synced = 1;
 	firstadj = 0;
+	update_time_sync_status(synced);
 	return (synced);
 }
 
-- 
2.4.5

