From 802b2ccf5257b840ed252a2bc1a6cb70a37b3f57 Mon Sep 17 00:00:00 2001
From: Brent Cook <busterb@gmail.com>
Date: Wed, 31 Dec 2014 22:39:58 -0600
Subject: [PATCH 14/14] add MAX macro

---
 src/usr.sbin/ntpd/ntpd.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/usr.sbin/ntpd/ntpd.h b/src/usr.sbin/ntpd/ntpd.h
index f8d6382..5ac0bbb 100644
--- a/src/usr.sbin/ntpd/ntpd.h
+++ b/src/usr.sbin/ntpd/ntpd.h
@@ -72,6 +72,10 @@
 #define	SENSOR_SCAN_INTERVAL		(5*60)
 #define	SENSOR_DEFAULT_REFID		"HARD"
 
+#ifndef MAX
+#define MAX(a, b) ((a) > (b) ? (a) : (b))
+#endif
+
 enum client_state {
 	STATE_NONE,
 	STATE_DNS_INPROGRESS,
-- 
1.9.1

