From 744376e0a488f7a0d594ea6b3d4b31ac684d4b7e Mon Sep 17 00:00:00 2001
From: Brent Cook <busterb@gmail.com>
Date: Mon, 8 Jun 2020 06:53:53 -0500
Subject: [PATCH 15/24] handle KERN_SECURELVL when available

---
 src/usr.sbin/ntpd/ntpd.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/usr.sbin/ntpd/ntpd.c b/src/usr.sbin/ntpd/ntpd.c
index eca3fa21e9b..5ec30d4fc38 100644
--- a/src/usr.sbin/ntpd/ntpd.c
+++ b/src/usr.sbin/ntpd/ntpd.c
@@ -21,7 +21,9 @@
 #include <sys/resource.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
+#ifdef KERN_SECURELVL
 #include <sys/sysctl.h>
+#endif
 #include <sys/wait.h>
 #include <sys/un.h>
 #include <netinet/in.h>
@@ -120,12 +122,14 @@ usage(void)
 int
 auto_preconditions(const struct ntpd_conf *cnf)
 {
+	int constraints, securelevel = 0;
+
+#ifdef KERN_SECURELVL
 	int mib[2] = { CTL_KERN, KERN_SECURELVL };
-	int constraints, securelevel;
 	size_t sz = sizeof(int);
-
 	if (sysctl(mib, 2, &securelevel, &sz, NULL, 0) == -1)
 		err(1, "sysctl");
+#endif
 	constraints = !TAILQ_EMPTY(&cnf->constraints);
 	return !cnf->settime && (constraints || cnf->trusted_peers ||
 	    conf->trusted_sensors) && securelevel == 0;
-- 
2.53.0

