From e33efdd820329c9232f657032cb8b900a8a50d6b Mon Sep 17 00:00:00 2001
From: Brent Cook <busterb@gmail.com>
Date: Mon, 20 Apr 2026 21:40:43 -0500
Subject: [PATCH 18/24] allow overiding the constraint path

---
 src/usr.sbin/ntpd/constraint.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/usr.sbin/ntpd/constraint.c b/src/usr.sbin/ntpd/constraint.c
index 5abaf27b279..e4b2c775697 100644
--- a/src/usr.sbin/ntpd/constraint.c
+++ b/src/usr.sbin/ntpd/constraint.c
@@ -376,7 +376,12 @@ priv_constraint_child(const char *pw_dir, uid_t pw_uid, gid_t pw_gid)
 
 #ifdef HAVE_LIBTLS
 	/* load CA certs before chroot() */
-	if ((conf->ca = tls_load_file(tls_default_ca_cert_file(),
+	if ((conf->ca = tls_load_file(
+#ifdef CONSTRAINT_CA
+	    CONSTRAINT_CA,
+#else
+	    tls_default_ca_cert_file(),
+#endif
 	    &conf->ca_len, NULL)) == NULL)
 		fatalx("failed to load constraint ca");
 #endif
-- 
2.53.0

