From 6aa45e8136a388a8c0821e9145e9989d031b9498 Mon Sep 17 00:00:00 2001
From: Brent Cook <busterb@gmail.com>
Date: Mon, 16 Nov 2020 02:22:07 -0600
Subject: [PATCH 20/21] remove cdefs from log.h (not clear why that's there)

---
 src/usr.sbin/ntpd/log.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/usr.sbin/ntpd/log.h b/src/usr.sbin/ntpd/log.h
index c3221c373e..2933d0e691 100644
--- a/src/usr.sbin/ntpd/log.h
+++ b/src/usr.sbin/ntpd/log.h
@@ -20,7 +20,6 @@
 #define LOG_H
 
 #include <stdarg.h>
-#include <sys/cdefs.h>
 
 #define LOG_TO_STDERR	(1<<0)
 #define LOG_TO_SYSLOG	(1<<1)
-- 
2.31.1

