From fd6d7d0cb7531d2863fe51aaaf4822a38fdae579 Mon Sep 17 00:00:00 2001
From: Brent Cook <busterb@gmail.com>
Date: Sun, 17 May 2026 07:53:48 -0500
Subject: [PATCH 25/27] if a pool is trusted, members of the pool are trusted

---
 src/usr.sbin/ntpd/ntp.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/usr.sbin/ntpd/ntp.c b/src/usr.sbin/ntpd/ntp.c
index 81c12e5a5e6..88a4e235a5d 100644
--- a/src/usr.sbin/ntpd/ntp.c
+++ b/src/usr.sbin/ntpd/ntp.c
@@ -635,6 +635,7 @@ ntp_dispatch_imsg_dns(void)
 					    log_ntp_addr(h), peer->addr_head.name);
 					npeer = new_peer();
 					npeer->weight = peer->weight;
+					npeer->trusted = peer->trusted;
 					npeer->query_addr4 = peer->query_addr4;
 					npeer->query_addr6 = peer->query_addr6;
 					h->next = NULL;
-- 
2.53.0

