--- a/src/usr.sbin/ntpd/ntpd.h	Wed Jan 28 16:02:41 2015
+++ b/src/usr.sbin/ntpd/ntpd.h	Wed Jan 28 15:55:40 2015
@@ -40,6 +40,10 @@
 #define	NTPD_USER	"_ntp"
 #endif
 
+#ifndef NTPD_CHROOT_DIR
+#define	NTPD_CHROOT_DIR	"/var/empty"
+#endif
+
 #ifndef	SYSCONFDIR
 #define	SYSCONFDIR	"/etc"
 #endif
--- a/src/usr.sbin/ntpd/ntp.c	Wed Jan 28 16:01:33 2015
+++ b/src/usr.sbin/ntpd/ntp.c	Wed Jan 28 15:57:31 2015
@@ -121,14 +121,14 @@
 	ntp_dns(pipe_dns, nconf, pw);
 	close(pipe_dns[1]);
 
-	if (stat(pw->pw_dir, &stb) == -1) {
-		fatal("privsep dir %s could not be opened", pw->pw_dir);
+	if (stat(NTPD_CHROOT_DIR, &stb) == -1) {
+		fatal("privsep dir %s could not be opened", NTPD_CHROOT_DIR);
 	}
 	if (stb.st_uid != 0 || (stb.st_mode & (S_IWGRP|S_IWOTH)) != 0) {
 		fatalx("bad privsep dir %s permissions: %o",
-		    pw->pw_dir, stb.st_mode);
+		    NTPD_CHROOT_DIR, stb.st_mode);
 	}
-	if (chroot(pw->pw_dir) == -1)
+	if (chroot(NTPD_CHROOT_DIR) == -1)
 		fatal("chroot");
 	if (chdir("/") == -1)
 		fatal("chdir(\"/\")");
